dace/autodiff/library/library.py

```diff
@@ -0,0 +1,295 @@
+# Copyright 2019-2025 ETH Zurich and the DaCe authors. All rights reserved.
+"""
+Dace library for autodiff
+
+Includes the BackwardPass library node, and the replacements for the python frontend
+"""
+from typing import Dict, Set, Optional
+import copy
+
+import dace
+import dace.library
+from dace import data, properties
+from dace.transformation import transformation as pm
+from dace.sdfg import SDFG, SDFGState, graph, nodes
+
+from dace.autodiff import backward_pass_generator as engine, analysis as autodiff_analysis
+from dace.autodiff.utils import init_grad
+from dace.util import in_edge_with_name
+from dace.transformation.passes.analysis import AccessSets
+
+
+@properties.make_properties
+class ParameterArray(data.Array):
```

Comment:
it's a nice solution, but can't it be in dace/data/autodiff_data.py or something? Let's keep that in mind for the next PR please.